### PR TITLE
Split starting game from started game

### PR DESCRIPTION
**Pull request by agustinesco**
Closes #[issue]

## Motivation

Whenever you entered any game you were prompt with a lot of errors (~600) .

THIS PR SHOULD BE MERGERD AND TESTED WITH [This PR](https://github.com/lambdaclass/game_backend/pull/58) FROM THE BACKEND REPO IN ORDER TO WORK

## Summary of changes

- Added new event of preparing game to the game event to init the game and change to the battle scene 
- Make every event that depends on game initialization to actually wait until the game is started

## How has this been tested?

Start a local server with the branch from the pr number 58 from the backend repo 

Enter a match from a localhost server and you should be just prompted with 9 errors that are unrelated to this pr

## Test additions / changes

List tests added/updated.

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

File: communication/messages.proto [hunk 1]
```diff
@@ -334,8 +335,8 @@ enum LobbyEventType {
     TYPE_UNSPECIFIED = 0;
     CONNECTED = 1;
     PLAYER_ADDED = 2;
-    GAME_STARTED = 3;
-    START_GAME = 4;
+    START_GAME = 3;
```
Comment (mFragaBA) [inline communication/messages.proto:338]: same as I commented in the backend PR. I think we are not using this one anymore

**Review by mFragaBA — CHANGES_REQUESTED** [review]
As talked with @saugon and @agustinesco, since we want to optimize scene loading times, we'll try keeping everything related to the battle in a single scene. That also means the preparing battle scene added in this PR since we can use that time to load the scene. 
For that we can introduce a new "preparing state" in the battle scene in which it doesn't listen for events. It just loads the scene and only when the game actually starts we can listen for new events

**Comment by mFragaBA** [comment]
I tried this on both an android phone and on unity and it's getting pretty laggy

https://github.com/lambdaclass/curse_of_myrra/assets/25107475/87ff2527-f733-4e0e-9e29-4c8de317645e

File: client/Assets/Scripts/Battle.cs [hunk 2]
```diff
@@ -102,6 +101,12 @@ void Update()
             SetAccumulatedTime();
             UpdateBattleState();
         }
+
+        if (LobbyConnection.Instance.gameStarted && !sendMovementStarted) {
+            sendMovementStarted = true;
+            float clientActionRate = SocketConnectionManager.Instance.serverTickRate_ms / 1000f;
+            InvokeRepeating("SendPlayerMovement", clientActionRate, clientActionRate);
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle.cs:108]:
```suggestion
            InvokeRepeating("SendPlayerMovement", 0, clientActionRate);
```

It's not necessary to wait the first time

**Review by ncontinanza — CHANGES_REQUESTED** [review]

**Review by ncontinanza — DISMISSED** [review]
Looking great!

**Review by mFragaBA — APPROVED** [review]
LGTM! The slowdown issue has been fixed!

**Review by ncontinanza — APPROVED** [review]

**Pull request merged**